### PR TITLE
Filter dags by clicking on tag

### LINKED ISSUE #8821: Add ability to filter dag list by clicking on specific tag
**Description**

tags was added in https://github.com/apache/airflow/pull/6489
it would be great if by clicking a tag it will show all the tags associated to this tag.
Similar to how filter tags search bar works just without the need to type.

### PR DESCRIPTION
**Pull request by Sangarshanan**
Closes #8821

Clicking on a tag to filter dags with that tag, a quicker alternative to searching

This my first pull request on airflow and is a minor feature, happy to make changes if anything needs to be updated or fixed ;)

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by mik-laj** [comment]
@zacharya19 Can you look at it?

File: airflow/www/templates/airflow/dags.html [hunk 1]
```diff
@@ -116,7 +116,9 @@ <h2>DAGs</h2>
 
               <div style="float: right; max-width: 70%; text-align: right; line-height: 160%;">
                 {% for tag in dag.tags | sort(attribute='name') %}
-                  <span class="label label-success" style="margin: 3px;">{{ tag.name }}</span>
+                  <a class="label label-success"
+                  href = "?tags={{ tag.name }}"
```
Comment (zacharya19) [inline airflow/www/templates/airflow/dags.html]:
```suggestion
                  href="?tags={{ tag.name }}"
```

**Review by zacharya19 — APPROVED** [review]
Nice!

**Comment by Sangarshanan** [comment]
Not sure why Kubernetes tests failed : (

**Comment by zacharya19** [comment]
@Sangarshanan It's not an issue with your code (obviously :) ), look like master is broken.

**Comment by Sangarshanan** [comment]
would you be able to help us here @mik-laj ?

**Comment by mik-laj** [comment]
@Sangarshanan Can you do rebase? Kubernetes and Quarantine tests are flaky. You can ignore it, but the others should work.

**Comment by Sangarshanan** [comment]
Everything except the Kubernetes tests passed @mik-laj

**Review by mik-laj — APPROVED** [review]

**Pull request merged**